itests/src/test/java/fi/metatavu/edelphi/test/ui/base/index/ChangePlanTestsBase.java

```diff
@@ -0,0 +1,179 @@
+  
+  @Test
+  public void testPlusToPlus() {
+    OffsetDateTime oldEnds = OffsetDateTime.now().plusDays(30);
+    testPlanChange(planMocker.getPlanId(1), "PLUS", "Plus", oldEnds, 1, 100, "PLUS", "Plus", 100, null, 30);
+    testPlanChange(planMocker.getPlanId(1), "PLUS", "Plus", oldEnds, 2, 200, "PLUS", "Plus", 180, null, 30);
+  }
+  
+  @Test
+  public void testPlusToPremiumNoCompensation() {
+    testPlanChange(planMocker.getPlanId(1), "PLUS", "Plus", null, 3, 100, "PREMIUM", "Premium", 200, null, null);
+    testPlanChange(planMocker.getPlanId(1), "PLUS", "Plus", null, 4, 200, "PREMIUM", "Premium", 360, null, null);
+  }
+  
+  @Test
+  public void testPlusToPremiumWithCompensation() {
+    OffsetDateTime oldEnds = OffsetDateTime.now().plusDays(30);
+    testPlanChange(planMocker.getPlanId(1), "PLUS", "Plus", oldEnds, 3, 100, "PREMIUM", "Premium", 200, 26.1d, null);
+    testPlanChange(planMocker.getPlanId(1), "PLUS", "Plus", oldEnds, 4, 200, "PREMIUM", "Premium", 360, 26.1d, null);
+  }
+  
+  @Test
+  public void testPremiumToPremium() {
+    OffsetDateTime oldEnds = OffsetDateTime.now().plusDays(30);
+    testPlanChange(planMocker.getPlanId(3), "PREMIUM", "Premium", oldEnds, 3, 100, "PREMIUM", "Premium", 200, null, 30);
+    testPlanChange(planMocker.getPlanId(3), "PREMIUM", "Premium", oldEnds, 4, 200, "PREMIUM", "Premium", 360, null, 30);
+  }
+  
+  @Test
+  public void testPremiumToPlus() {
+    updateUserSubscription(1l, "PREMIUM", null, Date.from(OffsetDateTime.now().plusDays(30).toInstant()));
+    updateUserPlan(1l, planMocker.getPlanId(3));
+    
+    login(ADMIN_EMAIL);
+    navigate(CHANGEPLAN_PAGE);
+
+    assertText("#changePlanGenericBlockContent .planFieldContainer:nth-of-type(1) .planHint", "You may not change your order to lower level when subscription is active");
+    assertInputDisabled("#changePlanGenericBlockContent .planFieldContainer:nth-of-type(1) input");
+    
+    assertText("#changePlanGenericBlockContent .planFieldContainer:nth-of-type(2) .planHint", "You may not change your order to lower level when subscription is active");
+    assertInputDisabled("#changePlanGenericBlockContent .planFieldContainer:nth-of-type(2) input");
+    
+    assertText("#changePlanGenericBlockContent .planFieldContainer:nth-of-type(3) .planHint", "Remaining days (30) of your order will be added to your new order");
+    assertInputEnabled("#changePlanGenericBlockContent .planFieldContainer:nth-of-type(3) input");
+    
+    assertText("#changePlanGenericBlockContent .planFieldContainer:nth-of-type(4) .planHint", "Remaining days (30) of your order will be added to your new order");
+    assertInputEnabled("#changePlanGenericBlockContent .planFieldContainer:nth-of-type(4) input");
+  }
+
+  private void testPlanChange(Long oldPlanId, String oldPlanType, String oldPlanName, OffsetDateTime fromEnds, int newPlanIndex, int newPlanDays, String newPlanType, String newPlanName, double newPlanPrice, Double compensation, Integer extraDays) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Method has 11 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00107)